test/test_genericprocess.py

```diff
@@ -0,0 +1,56 @@
+# pylint: disable=missing-docstring
+# -*- coding: utf-8 -*-
+
+"""
+Copyright 2019 ARM Limited
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import subprocess
+import unittest
+import platform
+import os
+
+from icetea_lib.IceteaManager import ExitCodes
+
+
+class GenericProcessTestcase(unittest.TestCase):
+    def test_quick_process(self):
+        # Windows doesn't have /bin/echo, use "cmd -c echo <test string>"
+        # from different configration
+        tc_cfg = os.path.join("test", "test_genericprocess_tc_cfg_unix.json")
+        if platform.system() == "Windows":
+            tc_cfg = os.path.join("test", "test_genericprocess_tc_cfg_windows.json")
+
+        # Run testcase
+        retcode = subprocess.call("python icetea.py --clean -s "
+                                  "--tc test_quick_process "
+                                  "--tcdir test --type process "
+                                  "--failure_return_value "
+                                  "--tc_cfg \"{}\"".format(tc_cfg),
+                                  shell=True)
+
+        # Check success
+        self.assertEqual(retcode, ExitCodes.EXIT_SUCCESS,
+                         "Non-success returncode returned. "
+                         "Allowed_platforms and platform_name "
+                         "broken.")
+
+        # Run with --clean to clean up
+        retcode = subprocess.call(
```

Comment:
Should be done in the class teardown function instead, because this will not get called if the assert above fails.

Reply:
Moved.